File: backend/api/models/book.py
from django.db import models

from api.models.author import Author


class Book(models.Model):
    name = models.CharField(max_length=255)
    description = models.TextField()
    image = models.ImageField(upload_to="book_images/", blank=True, null=True)
    author = models.ForeignKey(Author, on_delete=models.CASCADE, related_name="books")

    def __str__(self):
        return self.name

File: frontend/src/models/book.ts
import { Author } from "@/models";

export interface BookFromListRespose {
    id: number,
    name: string,
    description: string,
    image: string,
    author: number
}

export interface SingleBookResponse {
    id: number,
    name: string,
    description: string,
    image: string,
    author: Author
}